tests/ModelTest.php

```diff
@@ -657,161 +656,80 @@ public function testDates(): void
         $item = Item::create(['name' => 'sword']);
         $json = $item->toArray();
         $this->assertEquals($item->created_at->toISOString(), $json['created_at']);
+    }
 
-        /** @var User $user */
-        //Test with create and standard property
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => time()]);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => Date::now()]);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => 'Monday 8th August 2005 03:12:46 PM']);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => 'Monday 8th August 1960 03:12:46 PM']);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => '2005-08-08']);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => '1965-08-08']);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => new DateTime('2010-08-08')]);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => new DateTime('1965-08-08')]);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => new DateTime('2010-08-08 04.08.37')]);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => new DateTime('1965-08-08 04.08.37')]);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => new DateTime('2010-08-08 04.08.37.324')]);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
+    public static function provideDate(): \Generator
+    {
+        yield [time()];
```

Comment:
I suggest defining names for each data set:
```suggestion
        yield 'int timestamp' => [time()];
```